### ISSUE TITLE
[Bug]: Missing exports when using Rslib with Rspack 1.2.0-alpha.0

### ISSUE THREAD

**Issue by chenjiahan** [web-infra-dev/rslib#633]
### Version

```sh
System:
    OS: macOS 14.6.1
    CPU: (16) arm64 Apple M3 Max
    Memory: 2.71 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 131.0.6778.205
    Safari: 17.6
  npmPackages:
    @rslib/core: ^0.2.2 => 0.2.2
```

### Details

Overrides Rspack to [1.2.0-alpha.0](https://github.com/web-infra-dev/rspack/releases/tag/v1.2.0-alpha.0):

```json
"pnpm": {
  "overrides": {
    "@rspack/core": "1.2.0-alpha.0"
  }
},
```

Then the rsbuild-plugin-rem is broken:

![Image](https://github.com/user-attachments/assets/5a8f3ab3-5126-4d87-8eea-033acd64e1d5)

The `PLUGIN_REM_NAME` variable is missing from the output bundle:

![Image](https://github.com/user-attachments/assets/043fefd1-745f-4f7a-ae41-bfaaa1b1c096)

### Reproduce link

https://github.com/rspack-contrib/rsbuild-plugin-rem

### Reproduce Steps

1. clone repo
2. git checkout rslib_repro_1231
3. pnpm build
4. pnpm test

**Comment by fi3ework** [comment]
Yeah, it's because we lost `optimization.avoidEntryIife` which is disabled in 1.2.0-alpha.0 by default. `@rslib/core` has set the dependency of @rsbuild/core to `@rsbuild/core": "~1.1.8` so it's safe.

Rslib will add this configuration when `@rsbuild/core` 1.2.0 is released.

**Comment by chenjiahan** [comment]
Get 👌

**Comment by chenjiahan** [comment]
Fixed by https://github.com/web-infra-dev/rslib/releases/tag/v0.3.0
